src/entry.c

```diff
@@ -39,8 +41,18 @@
 static void xcb_xrm_append_char(xcb_xrm_entry_t *entry, xcb_xrm_entry_parser_state_t *state,
         const char str) {
     if (state->buffer_pos == NULL) {
-        memset(&(state->buffer), 0, sizeof(state->buffer));
+        FREE(state->buffer);
+        state->buffer = calloc(1, BUFFER_SIZE);
         state->buffer_pos = state->buffer;
+        if (state->buffer == NULL) {
+            /* Let's ignore this character and try again next time. */
+            return;
+        }
+    }
+
+    /* Increase the buffer if necessary. */
+    if ((state->buffer_pos - state->buffer) % (BUFFER_SIZE + 1) == BUFFER_SIZE) {
```

Comment:
> Thus my suggested fix is to replace the code here with (state->buffer_pos - state->buffer) % BUFFER_SIZE == BUFFER_SIZE - 1.

Ugh, this is literally what I had up until a second before committing when I thought "it'd look better this way". Oh well. :)